### ISSUE TITLE
Create local ci checks and github actions

### ISSUE THREAD

**Issue by chrishalbert** [zalando/gin-oauth2#78]
While working #76, I had issues running all of the tests locally. Also, my PR didn't abide to go doc standards, which I also didn't catch. I think this repo could benefit from 2 things:

1. **Local CI Checks** - This could be done using a `Makefile`, which would essentially be duplicating the checks locally so that a developer can verify their PR conforms to standards and the tests work. The `Makefile` could also automate the setup needed to run the tests locally.
2. **Github Actions** - I think the free account allows like 2k action minutes p/month before getting charged. I'm not sure what the owner is ok with, but maybe there could be a post check upon a merge to master. This would keep the usage minimal, even though it would be nice to have this run for each PR - easier said though when I'm not floating the bill 😄

**Comment by szuecs** [comment]
Yes thank you, I should create some gh action to lint and run tests.

**Comment by chrishalbert** [comment]
Or even just a way that _As a contributing developer, I can run tests and linting locally in the same manner that the CI checks run_.  On that note, how would I get a `.chimp-token` to get the `zalando_test.go` passing? Is there like a mocked server that I can point to for testing purposes? It seems like the CI checks run a number of services that require secrets which makes it hard to dev if not apart of the org.

**Comment by szuecs** [comment]
@chrishalbert yes likely we should just skip the zalando_test.go
